node.go

```diff
@@ -0,0 +1,335 @@
+	QueryRow(query string, args ...interface{}) *sql.Row
+	// QueryRowContext executes a statement that returns at most one row using given arguments.
+	QueryRowContext(ctx context.Context, query string, args ...interface{}) *sql.Row
+	// Prepare creates a prepared statement for later queries or executions.
+	// Multiple queries or executions may be run concurrently from the returned statement.
+	Prepare(query string) (*sql.Stmt, error)
+	// PrepareContext creates a prepared statement for later queries or executions.
+	// Multiple queries or executions may be run concurrently from the returned statement.
+	PrepareContext(ctx context.Context, query string) (*sql.Stmt, error)
+
+	// ----------------------------------------------------------------------------
+	// Connection
+	// ----------------------------------------------------------------------------
+
+	// DriverName returns the driver name used by this connector.
+	DriverName() string
+	// Ping verifies that the underlying connection is healthy.
+	Ping() error
+	// PingContext verifies that the underlying connection is healthy.
+	PingContext(ctx context.Context) error
+	// Close closes the underlying connection.
+	Close() error
+	// SetConnMaxLifetime sets the maximum amount of time a connection may be reused.
+	// Expired connections may be closed lazily before reuse.
+	SetConnMaxLifetime(duration time.Duration)
+	// SetMaxIdleConns sets the maximum number of connections in the idle connection pool.
+	SetMaxIdleConns(number int)
+	// SetMaxOpenConns sets the maximum number of open connections to the database.
+	SetMaxOpenConns(number int)
+	// EnableSavepoint activate PostgreSQL savepoints for nested transactions.
+	EnableSavepoint(enabled bool)
+	// Stats returns database statistics.
+	Stats() sql.DBStats
+
+	// ----------------------------------------------------------------------------
+	// Transaction
+	// ----------------------------------------------------------------------------
+
+	// Begin a new transaction.
+	Begin() (Node, error)
+	// Rollback the associated transaction.
+	Rollback() error
+	// Commit the associated transaction.
+	Commit() error
+
+	// ----------------------------------------------------------------------------
+	// System
+	// ----------------------------------------------------------------------------
+
+	// Tx returns the underlying transaction.
+	Tx() *sql.Tx
+	// DB returns the underlying connection.
+	DB() *sql.DB
+}
+
+type node struct {
+	driver           string
+	db               *sql.DB
+	tx               *sql.Tx
+	savePointID      string
+	savePointEnabled bool
+	nested           bool
+}
+
+// Connect to a database and verify with a ping.
+func Connect(driver string, dsn string) (Node, error) {
+	db, err := sql.Open(driver, dsn)
+	if err != nil {
+		return nil, err
+	}
+
+	err = db.Ping()
+	if err != nil {
+		// the connection has been opened within this function, we must close it
+		// on error.
+		_ = db.Close()
```

Comment:
We should wrap the `Close` error with `err` if its occurs instead of bypassing it silently.

Reply:
How you would do that easely?
Is it what you had in mind?

```go
thr := db.Close()
if thr != nil {
   err = errors.Wrap(err, thr.Error())
}
return nil, err
```

Reply:
@gillesfabio If you could give me some pointers, it would be highly appreciated :slightly_smiling_face: 
Thank you.